front-end/src/components/products/ProductForm.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import { InputText } from 'primereact/inputtext';
+import { Button } from 'primereact/button';
+import { Dropdown } from 'primereact/dropdown';
+import { useDispatch, useSelector } from 'react-redux';
+import { Product } from '../../features/product/types';
+import { addProduct, editProduct } from '../../features/product/productThunk';
+import { getSelectedProduct } from '../../features/product/selectors';
+import {
+    productCodeLabel,
+    productCodePlaceHolder,
+    productNameLabel,
+    productNamePlaceHolder,
+    productPackagingLabel,
+    productPackagingPlaceHolder,
+    saveLabel,
+} from '../../helpers/constants';
+
+export const ProductForm: React.FC = () => {
+    const dispatch = useDispatch();
+    const selectedProduct: Product | null = useSelector(getSelectedProduct);
+
+    const initialProductCode = selectedProduct ? selectedProduct.code : '';
+    const initialProductName = selectedProduct ? selectedProduct.name : '';
+    const initialPackaging = selectedProduct ? selectedProduct.packaging : '';
+
+    const [selectedPackaging, updatePackaging] = useState(initialPackaging);
+    const [productCode, updateProductCode] = useState(initialProductCode);
+    const [productName, updateProductName] = useState(initialProductName);
+
+    const packagings = [
```

Comment:
The units are always represented in the small letters. There some exceptions where capitals also are used but those exceptions don't fall here.

Reply:
Sure lets define it how to display, may be I would keep like these as of now else my mock apis wont work.